script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -110,45 +134,219 @@ class PublishPluginCommand extends PluginCommand {
       remoteUrl = await _verifyRemote(remote);
     }
     _print('Local repo is ready!');
+    if (argResults[_dryRunFlag] as bool) {
+      _print('===============  DRY RUN ===============');
+    }
 
-    final Directory packageDir = _getPackageDir(package);
-    await _publishPlugin(packageDir: packageDir);
+    bool successful;
+    if (publishAllChanged) {
+      successful = await _publishAllChangedPackages(
+          remote: remote,
+          remoteUrl: remoteUrl,
+          shouldPushTag: shouldPushTag,
+          baseGitDir: baseGitDir);
+    } else {
+      successful = await _publishAndTagPackage(
+          packageDir: _getPackageDir(package),
+          remote: remote,
+          remoteUrl: remoteUrl,
+          shouldPushTag: shouldPushTag);
+    }
+    await _finish(successful);
+  }
+
+  Future<bool> _publishAllChangedPackages(
+      {String remote,
+      String remoteUrl,
+      bool shouldPushTag,
+      GitDir baseGitDir}) async {
```

Comment:
Optional nit: add trailing commas here and below to improve formatting

Reply:
done